template_handler/cdk.py

```diff
@@ -0,0 +1,218 @@
+import aws_cdk
+import importlib.machinery
+import importlib.util
+import logging
+import os
+import pathlib
+import shutil
+import subprocess
+import yaml
+from botocore.credentials import Credentials
+from pathlib import Path
+from sceptre import exceptions
+from sceptre.template_handlers import TemplateHandler
+from typing import Dict
+
+
+class CDK(TemplateHandler):
+    """
+    A template handler for AWS CDK templates. Using this will allow Sceptre to work with the AWS CDK to
+    build and package a CDK template and deploy it with Sceptre.
+    """
+
+    class PrefixLoggerAdapter(logging.LoggerAdapter):
+        """
+        This logger adapter expects to be passed in a dict-like object with a
+        'prefix' key, whose value is prefixed to the log message.
+        """
+        def process(self, msg, kwargs):
+            return f'{self.extra["prefix"]} - {msg}', kwargs
+
+    def __init__(self, *args, **kwargs):
+        super(CDK, self).__init__(*args, **kwargs)
+
+    def schema(self):
+        """
+        Return a JSON schema of the properties that this template handler requires.
+        Reference: https://github.com/Julian/jsonschema
+        """
+        return {
+            "type": "object",
+            "properties": {
+                "path": {"type": "string"},
+                "context": {"type": "object"}
+            },
+            "required": [
+                "path"
+            ]
+        }
+
+    def _get_envs(self) -> Dict[str, str]:
+        """Obtains the environment variables to pass to the subprocess.
+
+        Sceptre can assume roles, profiles, etc... to connect to AWS for a given stack. This is
+        very useful. However, we need that SAME connection information to carry over to SAM when we
+        invoke it. The most precise way to do this is to use the same session credentials being used
+        by Sceptre for other stack operations. This method obtains those credentials and sets them
+        as environment variables that are passed to the subprocess and will, in turn, be used by
+        SAM CLI.
+
+        The environment variables dict created by this method will inherit all existing
+        environment variables in the current environment, but the AWS connection environment
+        variables will be overridden by the ones for this stack.
+
+        Returns:
+            The dictionary of environment variables.
+        """
+        envs = os.environ
+        # Set aws environment variables specific to whatever AWS configuration has been set on the
+        # stack's connection manager.
+        credentials: Credentials = self.connection_manager._get_session(
```

Comment:
Side-note, I've been meaning to make an issue to expose this method as a "public" method on the connection manager, since this is the only way to explicitly get a session from the stack's connection manager and it's technically "private".